system_modules/napaudio/src/audio/node/levelmeternode.cpp

```diff
@@ -61,43 +52,55 @@ namespace nap
 
 			if (inputBuffer == nullptr)
 				return;
-
-			for (auto& sample : *inputBuffer) {
-				mBuffer[mIndex++] = sample;
-				if (mIndex == mBuffer.size()) {
-					mIndex = 0;
-					mDirty.set();
-				}
+            
+			switch (mType)
+			{
+				case PEAK:
+					for (auto& sample : *inputBuffer)
+					{
+						// keep track of peak, sample by sample
+						float newValue = sample;
```

Comment:
For future reference, local variables are never `camelCase`, they are separated using an underscore: `float new_value `,  this allows you to differentiate between arguments and local variables, which is currently not the case. This is part of the official style guide.